[Bug]: `gradlew clean` involves external transform generation and fails

### What happened?

Repro:

```
 gradlew :sdks:python:cleanPython
To honour the JVM settings for this build a single-use Daemon process will be forked. For more on this, please refer to https://docs.gradle.org/8.4/userguide/gradle_daemon.html#sec:disabling_the_daemon in the Gradle documentation.
Daemon will be stopped at the end of the build 
Configuration on demand is an incubating feature.

> Task :sdks:python:cleanPython FAILED
b''
Traceback (most recent call last):
  File "setup.py", line 260, in generate_external_transform_wrappers
    subprocess.run([
  File "/home/valentyn/.pyenv/versions/3.8.18/lib/python3.8/subprocess.py", line 516, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/home/valentyn/projects/beam/beam2/beam/build/gradleenv/1922375555/bin/python', '/home/valentyn/projects/beam/beam2/beam/sdks/python/gen_xlang_wrappers.py', '--cleanup', '--transforms-config-source', '/home/valentyn/projects/beam/beam2/beam/sdks/standard_external_transforms.yaml']' returned non-zero exit status 1.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "setup.py", line 300, in <module>
    generate_external_transform_wrappers()
  File "setup.py", line 269, in generate_external_transform_wrappers
    raise RuntimeError(
RuntimeError: ('Could not generate external transform wrappers due to error: %s', b'Traceback (most recent call last):\n  File "/home/valentyn/projects/beam/beam2/beam/sdks/python/gen_xlang_wrappers.py", line 35, in <module>\n    import yaml\nModuleNotFoundError: No module named \'yaml\'\n')

FAILURE: Build failed with an exception.
```


### Issue Priority

Priority: 3 (minor)

### Issue Components

- [X] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam YAML
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

--- a/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy
+++ b/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy
@@ -3036,7 +3036,7 @@ class BeamModulePlugin implements Plugin<Project> {
           project.exec {
             executable 'sh'
             args '-c', "if [ -e ${activate} ]; then " +
-                ". ${activate} && cd ${pythonRootDir} && python setup.py clean; " +
+                ". ${activate} && cd ${pythonRootDir} && pip install pyyaml jinja2 && python setup.py clean; " +
                 "fi"
           }
           project.delete project.buildDir     // Gradle build directory

